.github/workflows/ci.yml

```diff
@@ -1,34 +1,47 @@
-name: Build and Test Spring Boot App
+name: Build, Test & SonarCloud
 
 on:
   push:
     branches:
       - main
       - develop
       - master
-      - release/*
-      - feature/*
-      - hotfix/*
+      - 'release/*'
+      - 'feature/*'
+      - 'hotfix/*'
   pull_request:
-    branches:
-      - develop
-      - master
-      - main
+    branches: [develop, master, main]
 
 jobs:
   build:
     runs-on: ubuntu-latest
-    
+
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
       - name: Set up JDK 17
         uses: actions/setup-java@v3
         with:
-          distribution: 'temurin'
-          java-version: '17'
-          cache: 'maven'
+          distribution: temurin
+          java-version: 17
+          cache: maven
 
-      - name: Build and run tests
-        run: mvn clean install
+      - name: Build & Test
+        run: ./mvnw -B -Pcoverage clean verify
+
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarqube-scan-action@v5.2.0
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: https://sonarcloud.io
+        with:
+          args: >
+            -Dsonar.organization=usdot-fhwa-stol
+            -Dsonar.projectKey=usdot-fhwa-stol_c2c-mvt
+            -Dsonar.java.binaries=target
+            -Dsonar.coverage.jacoco.xmlReportPaths=target/site/jacoco/jacoco.xml
```

Comment:
I do not see the coverage report and code smell report in the sonarcloud: https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&pullRequest=6&id=usdot-fhwa-stol_c2c-mvt

Reply:
Yes,  Sonar scan new-code scope compares this PR’s diff with the default branch that is develop. Since the PR doesn’t change any core code, the coverage and code-smell reports don’t appear here; you’ll see them on the develop branch once merged. After that each time on the PR which has any new issues it shows up if not it says passed.